core/embed/rust/src/ui/model_tr/component/pin.rs

```diff
@@ -0,0 +1,149 @@
+// Almost a copy of `page.rs` without the unnecessary stuff and with some extra
+// logic.
+
+use crate::ui::{
+    component::{Component, ComponentExt, Event, EventCtx, Pad, PageMsg, Paginate},
+    geometry::{Insets, Rect},
+};
+
+use super::{theme, Button, ButtonMsg, ButtonPos};
+use heapless::String;
+
+pub struct PinPage<T> {
+    content: T,
+    pad: Pad,
+    prev: Button<&'static str>,
+    next: Button<&'static str>,
+    accept_pin: Button<&'static str>,
+    cancel_pin: Button<&'static str>,
+    ok: Button<&'static str>,
+    pin_counter: u8,
+    pin_buffer: String<50>,
+}
+
+// TODO: somehow show the length of the PIN
+// TODO: somehow allow for viewing the PIN
+// TODO: somehow allow for deleting the digits in PIN
+
+impl<T> PinPage<T>
+where
+    T: Paginate,
```

Comment:
No need to involve pagination or inner `T` component at all.

Reply:
In [71740a4](https://github.com/trezor/trezor-firmware/pull/2233/commits/71740a44445495b4d36aca38074c199ce6fcffaf) I got completely rid of the pagination, currently, I am showing the digit on screen using `display::text`